src/client.js

```diff
@@ -461,10 +461,25 @@ function SugarApi(args) {
                 };
 
                 if (!('crosstab' in window) || !crosstab.supported) {
+
+                    // prevents concurrent events from multiple tabs asking for a
+                    // refresh token
+                    if (_refreshingToken) {
+                        return;
+                    }
+                    _refreshingToken = true;
                     self.login(null, {refresh: true}, {
-                        complete: refreshCallbacks.complete,
-                        success: refreshCallbacks.success,
-                        error: refreshCallbacks.error
+                        complete: function() {
+                            refreshCallbacks.complete;
```

Comment:
I believe, you should call the function here and below
```suggestion
                            refreshCallbacks.complete(param1, param2);
```

looks like your `refreshCallbacks.complete` is not called here.